go/vt/sqlparser/token.go

```diff
@@ -965,16 +993,28 @@ func (tkn *Tokenizer) Scan() (int, []byte) {
 				return NE, nil
 			}
 			return int(ch), nil
-		case '\'', '"':
+		case contains(tkn.stringLiteralQuotes, ch):
 			return tkn.scanString(ch, STRING)
-		case '`':
-			return tkn.scanLiteralIdentifier()
+		case contains(tkn.identifierQuotes, ch):
+			return tkn.scanLiteralIdentifier(ch)
 		default:
 			return LEX_ERROR, []byte{byte(ch)}
 		}
 	}
 }
 
+// contains searches the specified |slice| for the target |x|, and returns the same value of |x| if it is found. The
+// target value is returned, instead of a boolean response, so that this function can be directly used inside the
+// switch statement above that switches on a uint16 value.
+func contains(slice []uint16, x uint16) uint16 {
```

Comment:
This is on enough of a hot path we should put these in a map instead, I think

Or even a static array (one array for each mode)